{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Defaulting to user installation because normal site-packages is not writeable\n",
      "Collecting pyChatGPT\n",
      "  Downloading pyChatGPT-0.4.3.3-py3-none-any.whl (23 kB)\n",
      "Collecting undetected-chromedriver>=3.2.1\n",
      "  Downloading undetected-chromedriver-3.4.6.tar.gz (61 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m61.3/61.3 kB\u001b[0m \u001b[31m1.4 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25hCollecting markdownify>=0.11.6\n",
      "  Downloading markdownify-0.11.6-py3-none-any.whl (16 kB)\n",
      "Collecting beautifulsoup4<5,>=4.9\n",
      "  Downloading beautifulsoup4-4.12.2-py3-none-any.whl (142 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m143.0/143.0 kB\u001b[0m \u001b[31m3.2 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: six<2,>=1.15 in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from markdownify>=0.11.6->pyChatGPT) (1.16.0)\n",
      "Collecting selenium>=4.0.0\n",
      "  Downloading selenium-4.8.3-py3-none-any.whl (6.5 MB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.5/6.5 MB\u001b[0m \u001b[31m11.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hRequirement already satisfied: requests in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from undetected-chromedriver>=3.2.1->pyChatGPT) (2.28.2)\n",
      "Collecting websockets\n",
      "  Downloading websockets-11.0.1-cp311-cp311-macosx_10_9_x86_64.whl (120 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m120.8/120.8 kB\u001b[0m \u001b[31m4.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hCollecting soupsieve>1.2\n",
      "  Downloading soupsieve-2.4-py3-none-any.whl (37 kB)\n",
      "Requirement already satisfied: urllib3[socks]~=1.26 in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from selenium>=4.0.0->undetected-chromedriver>=3.2.1->pyChatGPT) (1.26.15)\n",
      "Collecting trio~=0.17\n",
      "  Downloading trio-0.22.0-py3-none-any.whl (384 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m384.9/384.9 kB\u001b[0m \u001b[31m6.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m00:01\u001b[0m\n",
      "\u001b[?25hCollecting trio-websocket~=0.9\n",
      "  Downloading trio_websocket-0.10.2-py3-none-any.whl (17 kB)\n",
      "Requirement already satisfied: certifi>=2021.10.8 in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from selenium>=4.0.0->undetected-chromedriver>=3.2.1->pyChatGPT) (2022.12.7)\n",
      "Requirement already satisfied: charset-normalizer<4,>=2 in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from requests->undetected-chromedriver>=3.2.1->pyChatGPT) (3.1.0)\n",
      "Requirement already satisfied: idna<4,>=2.5 in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from requests->undetected-chromedriver>=3.2.1->pyChatGPT) (3.4)\n",
      "Requirement already satisfied: attrs>=19.2.0 in /Users/eotenen/Library/Python/3.11/lib/python/site-packages (from trio~=0.17->selenium>=4.0.0->undetected-chromedriver>=3.2.1->pyChatGPT) (22.2.0)\n",
      "Collecting sortedcontainers\n",
      "  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)\n",
      "Collecting async-generator>=1.9\n",
      "  Downloading async_generator-1.10-py3-none-any.whl (18 kB)\n",
      "Collecting outcome\n",
      "  Downloading outcome-1.2.0-py2.py3-none-any.whl (9.7 kB)\n",
      "Collecting sniffio\n",
      "  Downloading sniffio-1.3.0-py3-none-any.whl (10 kB)\n",
      "Collecting exceptiongroup\n",
      "  Using cached exceptiongroup-1.1.1-py3-none-any.whl (14 kB)\n",
      "Collecting wsproto>=0.14\n",
      "  Downloading wsproto-1.2.0-py3-none-any.whl (24 kB)\n",
      "Collecting PySocks!=1.5.7,<2.0,>=1.5.6\n",
      "  Using cached PySocks-1.7.1-py3-none-any.whl (16 kB)\n",
      "Collecting h11<1,>=0.9.0\n",
      "  Downloading h11-0.14.0-py3-none-any.whl (58 kB)\n",
      "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m58.3/58.3 kB\u001b[0m \u001b[31m1.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
      "\u001b[?25hInstalling collected packages: sortedcontainers, websockets, soupsieve, sniffio, PySocks, outcome, h11, exceptiongroup, async-generator, wsproto, trio, beautifulsoup4, trio-websocket, markdownify, selenium, undetected-chromedriver, pyChatGPT\n",
      "\u001b[33m  DEPRECATION: undetected-chromedriver is being installed using the legacy 'setup.py install' method, because it does not have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change. A possible replacement is to enable the '--use-pep517' option. Discussion can be found at https://github.com/pypa/pip/issues/8559\u001b[0m\u001b[33m\n",
      "\u001b[0m  Running setup.py install for undetected-chromedriver ... \u001b[?25ldone\n",
      "\u001b[?25hSuccessfully installed PySocks-1.7.1 async-generator-1.10 beautifulsoup4-4.12.2 exceptiongroup-1.1.1 h11-0.14.0 markdownify-0.11.6 outcome-1.2.0 pyChatGPT-0.4.3.3 selenium-4.8.3 sniffio-1.3.0 sortedcontainers-2.4.0 soupsieve-2.4 trio-0.22.0 trio-websocket-0.10.2 undetected-chromedriver-3.4.6 websockets-11.0.1 wsproto-1.2.0\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip available: \u001b[0m\u001b[31;49m22.3.1\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m23.0.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpython3.11 -m pip install --upgrade pip\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!pip3 install -U pyChatGPT"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "URLError",
     "evalue": "<urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)>",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mSSLCertVerificationError\u001b[0m                  Traceback (most recent call last)",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:1348\u001b[0m, in \u001b[0;36mAbstractHTTPHandler.do_open\u001b[0;34m(self, http_class, req, **http_conn_args)\u001b[0m\n\u001b[1;32m   1347\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m-> 1348\u001b[0m     h\u001b[39m.\u001b[39;49mrequest(req\u001b[39m.\u001b[39;49mget_method(), req\u001b[39m.\u001b[39;49mselector, req\u001b[39m.\u001b[39;49mdata, headers,\n\u001b[1;32m   1349\u001b[0m               encode_chunked\u001b[39m=\u001b[39;49mreq\u001b[39m.\u001b[39;49mhas_header(\u001b[39m'\u001b[39;49m\u001b[39mTransfer-encoding\u001b[39;49m\u001b[39m'\u001b[39;49m))\n\u001b[1;32m   1350\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mOSError\u001b[39;00m \u001b[39mas\u001b[39;00m err: \u001b[39m# timeout error\u001b[39;00m\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py:1282\u001b[0m, in \u001b[0;36mHTTPConnection.request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1281\u001b[0m \u001b[39m\"\"\"Send a complete request to the server.\"\"\"\u001b[39;00m\n\u001b[0;32m-> 1282\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_send_request(method, url, body, headers, encode_chunked)\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py:1328\u001b[0m, in \u001b[0;36mHTTPConnection._send_request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1327\u001b[0m     body \u001b[39m=\u001b[39m _encode(body, \u001b[39m'\u001b[39m\u001b[39mbody\u001b[39m\u001b[39m'\u001b[39m)\n\u001b[0;32m-> 1328\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mendheaders(body, encode_chunked\u001b[39m=\u001b[39;49mencode_chunked)\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py:1277\u001b[0m, in \u001b[0;36mHTTPConnection.endheaders\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1276\u001b[0m     \u001b[39mraise\u001b[39;00m CannotSendHeader()\n\u001b[0;32m-> 1277\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_send_output(message_body, encode_chunked\u001b[39m=\u001b[39;49mencode_chunked)\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py:1037\u001b[0m, in \u001b[0;36mHTTPConnection._send_output\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1036\u001b[0m \u001b[39mdel\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m_buffer[:]\n\u001b[0;32m-> 1037\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msend(msg)\n\u001b[1;32m   1039\u001b[0m \u001b[39mif\u001b[39;00m message_body \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m   1040\u001b[0m \n\u001b[1;32m   1041\u001b[0m     \u001b[39m# create a consistent interface to message_body\u001b[39;00m\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py:975\u001b[0m, in \u001b[0;36mHTTPConnection.send\u001b[0;34m(self, data)\u001b[0m\n\u001b[1;32m    974\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mauto_open:\n\u001b[0;32m--> 975\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mconnect()\n\u001b[1;32m    976\u001b[0m \u001b[39melse\u001b[39;00m:\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/http/client.py:1454\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1452\u001b[0m     server_hostname \u001b[39m=\u001b[39m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mhost\n\u001b[0;32m-> 1454\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39msock \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_context\u001b[39m.\u001b[39;49mwrap_socket(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msock,\n\u001b[1;32m   1455\u001b[0m                                       server_hostname\u001b[39m=\u001b[39;49mserver_hostname)\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py:517\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[0;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[1;32m    511\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mwrap_socket\u001b[39m(\u001b[39mself\u001b[39m, sock, server_side\u001b[39m=\u001b[39m\u001b[39mFalse\u001b[39;00m,\n\u001b[1;32m    512\u001b[0m                 do_handshake_on_connect\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m,\n\u001b[1;32m    513\u001b[0m                 suppress_ragged_eofs\u001b[39m=\u001b[39m\u001b[39mTrue\u001b[39;00m,\n\u001b[1;32m    514\u001b[0m                 server_hostname\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m, session\u001b[39m=\u001b[39m\u001b[39mNone\u001b[39;00m):\n\u001b[1;32m    515\u001b[0m     \u001b[39m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[1;32m    516\u001b[0m     \u001b[39m# ctx._wrap_socket()\u001b[39;00m\n\u001b[0;32m--> 517\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49msslsocket_class\u001b[39m.\u001b[39;49m_create(\n\u001b[1;32m    518\u001b[0m         sock\u001b[39m=\u001b[39;49msock,\n\u001b[1;32m    519\u001b[0m         server_side\u001b[39m=\u001b[39;49mserver_side,\n\u001b[1;32m    520\u001b[0m         do_handshake_on_connect\u001b[39m=\u001b[39;49mdo_handshake_on_connect,\n\u001b[1;32m    521\u001b[0m         suppress_ragged_eofs\u001b[39m=\u001b[39;49msuppress_ragged_eofs,\n\u001b[1;32m    522\u001b[0m         server_hostname\u001b[39m=\u001b[39;49mserver_hostname,\n\u001b[1;32m    523\u001b[0m         context\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m,\n\u001b[1;32m    524\u001b[0m         session\u001b[39m=\u001b[39;49msession\n\u001b[1;32m    525\u001b[0m     )\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py:1075\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[0;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[1;32m   1074\u001b[0m             \u001b[39mraise\u001b[39;00m \u001b[39mValueError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m-> 1075\u001b[0m         \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mdo_handshake()\n\u001b[1;32m   1076\u001b[0m \u001b[39mexcept\u001b[39;00m (\u001b[39mOSError\u001b[39;00m, \u001b[39mValueError\u001b[39;00m):\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/ssl.py:1346\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[0;34m(self, block)\u001b[0m\n\u001b[1;32m   1345\u001b[0m         \u001b[39mself\u001b[39m\u001b[39m.\u001b[39msettimeout(\u001b[39mNone\u001b[39;00m)\n\u001b[0;32m-> 1346\u001b[0m     \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_sslobj\u001b[39m.\u001b[39;49mdo_handshake()\n\u001b[1;32m   1347\u001b[0m \u001b[39mfinally\u001b[39;00m:\n",
      "\u001b[0;31mSSLCertVerificationError\u001b[0m: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mURLError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32mUntitled-1.ipynb Cell 2\u001b[0m in \u001b[0;36m4\n\u001b[1;32m      <a href='vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W3sdW50aXRsZWQ%3D?line=0'>1</a>\u001b[0m \u001b[39mfrom\u001b[39;00m \u001b[39mpyChatGPT\u001b[39;00m \u001b[39mimport\u001b[39;00m ChatGPT\n\u001b[1;32m      <a href='vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W3sdW50aXRsZWQ%3D?line=2'>3</a>\u001b[0m session_token \u001b[39m=\u001b[39m \u001b[39m'\u001b[39m\u001b[39meyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..A6rwlPIgjgI-xWGD.Bvbl71-Rkaa1-FfA9hEAV4PvoIA0P95ZaUYIzoQREaHGEXWrjP0JzTsryEH8extiZp_37AcEL2wZxmLt6-08Er7wOuddHDi6krfb1uV_ZboyM-tXEsghUkssjMehZMLeHkAZQ_tZx19IGq8U3IO1h_-nGHEfvnRs5PPrzft2dg3AmT7iTdZS36GSmS0yra1xQDGCvJwxGPiYPZEmtO5Di5Tew0nj48c4JqPqFkRISDADTJ6KL4sehnLw_2-j4eHOghnnht8sbnL8lUQ8Ytf9-hSL_GX8GwdGZEyQijfvDl4NeGlPZbmnZNryZG1_7a6h_V0jxU1Nyc1CIJ20jAn-b8joEvL7CIfXeM862IF6yC4uzXhvRNMaYYKLviVz5haB4b6t-g37ZzlMfhOR6Pev2jNeSTfuIEt4UhJoC8i5d6aoxumyd4PmsFwNrMuiCddT28cAmfuMPSW0KYE-K-Yz23k34Ua16fZuHBH9gkYfRt1zr2_zLwWygR74sOMQO2xxEEUfNcntiKVWDfaEZaS4EZJAfGEIJ-uZHI5YehabtEAgKRiVA937sYY7OZaQ12b3FntR-a_gIjQFnpX9DnnpjUlj-dLhiKK5WFSbxLgFwIoide4uu4w5LaJLPZAwGK7tLypTO7Z2TFvSyue4UF_Fn00u_nftAH4EH3TvfmvyDNZy7zkifqyupQ2FjFwjewpVTiO-l7ZLaU9F5MozpyvYwF0iH1whPrJfiyU4D0VTRIT9sNGI6hQn_C7b21F45obpZ6hdljPOUwwZR5LNdoknKtoHRL47NT-bpoAfza0ZnspuARPSrVp66c--CaJsPHGePjYSHSxECDz7Vn9ZPhJ_Nqjo_4PSWv7R0Lu_opFa9ka8kiFM4CqzgRZJjOvgzZQUNX1mw74WivCXZwAB4qyXhpywh0mIPkmxUXLOam6SopxWY9FL8673nJmIDFy4O6940auXU5mi4rLLIjqSULkLI0bOpdbZkLK5r7RBzsJXXOoYEEh0vFTYMmFjRLdVA58NKD2Pm5zOl3Lg1E45Xp7uycENFaVmO8-krLX4--CEX3I8-UjdhBvxgx2SRvdP2TRxGRC8OjVnpTSWGqnpSU-TcAjJTUBsBhQtZ4lU8zynGPzsmLPPObRAfS7WQdvrvAz0tTZSL1OJkCZnBb_NjpGzMRzwkhwJiohV76efaDSIW7KyYe_dVXL44hL783GPMGEp0L6ahMgydeWAFMLXpoaO366r56hMLyM9VlW8R3BfcmHrbnuqtIevYA5wEYtIqfLQ-gFg3ggGI8XyPg_mK4l-F0Di1OxpWcKsd3fSPspbeIEvWLFHEFwYTUWynedTwwyNY3-W15Lq8BI3uqwYI4mjKZ3nIkrS21U9AIDKVcnDVBRVj3tsFmlgNUCBAs5vykkbFSft2ksX1gyLzriHLl0aTWq2nE4OyAEhF0hUy2gZcdnctd2E02QMp4_K-KiyiOgqJHes_qKKQpenJCwKX8k_opd7zUA4BFTcca1W3R5ozLRAbLo2mQnUMuzuihU0sPZTTKnKKjTAz9roYcIvY7wpauETvGQHiypmAM_cPoxfXVMBuLZLfPq3a4t66x5ITGFcM6sjEt7Eyxo-UVzHNzxUhaAay_JccmSf0YoiAATjH4ONAGDjUleMij1nDTOS8YuZ1-x0L3C2JUDWVywCXf9Dgb2Y5_Qke_QnFl0cB-OoYNwH2vJ4mGSUHjlGZmBZDGNDGIWLXLhDiZfhNdDgeP7m7E7yQzRHVw0Hp-8R3W5Hl338ejFhjky4n0DW8SoU7aoR8GxPWBkgG7ST7KjKCHsst2iO5b6A_mkr8Mlg9GNN0x_5PaCvtcYrdPzTFPx0vRtOGjsEnlIXaOvVCEUl3dfHF7rFGV6ssYuKFJfMw94RTQhkoYNb2P7XeLJHwZK1SM8OgB3Q49r1yu0mUrS9U9stzDllLgAW6fS3xjhOMfVG4oVYmaJgz-FzOyJxSnaBigoMipLuON_nIUo7Y3Pt8nVeYunZSSYNzT0IfVbIL-HAHaMwj-kO0hmcYOmFdaRxlbiiMIqIQaCQymhXyCbNdDj5ltZBZNTSTFXrqBVt0k2ZqNSMQ_rVHV2E3p17zGZd47f6KBpkU0s9iGcp4wR1dp4bOe_Ca-zWiFt_jaQFG3YjWth55KS865mXRNcQKK4RTPEWWqRQv3yVrQUYWUojK92XOnvGpfce4wA1cPPykd8lVPNa_mSsrAUNAw-nwFjTEIdLyYKDCrGcjxUhCsnaNGi3oDbI7KrajubWmHkd0fR5E7xbylhS-QBJmGD_QmsG_gLqWttKEDWOfiCRL66ii_SYdV6LDKqSTuBYwqkJE-tDgu3qjU_RoW83Ma4Y4szNMVG-69jJDWBV1VN80f-gxWhhVW7yhvlewSVFar4yY0J4uOeNEQHQX0qV8IZMkxmMmPTTB1wwasBBq0tTxqPQ78G-VrKyhcd54E9S3REgeNl5zMPhxS1tEHMdpQQqONrSqKb4MFQGJlAYGKAZ_H4m61qcgcYwKydnPtIblLmFIwiAX_AbOJDIrpKdx_tKWyCn3Linj1kB018JgyJoYwhdYwCOurY0AaStMu6fLvjix2hlGwXFTXgzILmDCqTvLgu7nMVnxXVJnQB-TdeXYzO_QFAwRqG6JemPwiuP6r6aByv0SlboeE9aOAiRUr4DGYmIcZzRBF2xZy68UojanLKkvGKXQJyqhTlbUNx0aPHdqV8oKOKQpXxU4ob-j18Gc_xJm7WFDRZQF8_IHX82BlVyliUumPPBYOcCxSQ.2R8OfzEpLlaqUC5j5BTnSw\u001b[39m\u001b[39m'\u001b[39m  \u001b[39m# `__Secure-next-auth.session-token` cookie from https://chat.openai.com/chat\u001b[39;00m\n\u001b[0;32m----> <a href='vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W3sdW50aXRsZWQ%3D?line=3'>4</a>\u001b[0m api \u001b[39m=\u001b[39m ChatGPT(session_token)  \u001b[39m# auth with session token\u001b[39;00m\n\u001b[1;32m      <a href='vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W3sdW50aXRsZWQ%3D?line=4'>5</a>\u001b[0m api \u001b[39m=\u001b[39m ChatGPT(session_token, conversation_id\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39msome-random-uuid\u001b[39m\u001b[39m'\u001b[39m)  \u001b[39m# specify conversation id\u001b[39;00m\n\u001b[1;32m      <a href='vscode-notebook-cell:Untitled-1.ipynb?jupyter-notebook#W3sdW50aXRsZWQ%3D?line=5'>6</a>\u001b[0m api \u001b[39m=\u001b[39m ChatGPT(session_token, proxy\u001b[39m=\u001b[39m\u001b[39m'\u001b[39m\u001b[39mhttps://proxy.example.com:8080\u001b[39m\u001b[39m'\u001b[39m)  \u001b[39m# specify proxy\u001b[39;00m\n",
      "File \u001b[0;32m~/Library/Python/3.11/lib/python/site-packages/pyChatGPT/pyChatGPT.py:126\u001b[0m, in \u001b[0;36mChatGPT.__init__\u001b[0;34m(self, session_token, conversation_id, auth_type, email, password, login_cookies_path, captcha_solver, solver_apikey, proxy, chrome_args, moderation, verbose)\u001b[0m\n\u001b[1;32m    123\u001b[0m         subprocess\u001b[39m.\u001b[39mrun([\u001b[39m'\u001b[39m\u001b[39mffdl\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39minstall\u001b[39m\u001b[39m'\u001b[39m])\n\u001b[1;32m    124\u001b[0m     os\u001b[39m.\u001b[39menviron[\u001b[39m'\u001b[39m\u001b[39mPATH\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m+\u001b[39m\u001b[39m=\u001b[39m os\u001b[39m.\u001b[39mpathsep \u001b[39m+\u001b[39m ffdl\u001b[39m.\u001b[39mffmpeg_dir\n\u001b[0;32m--> 126\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m__init_browser()\n\u001b[1;32m    127\u001b[0m weakref\u001b[39m.\u001b[39mfinalize(\u001b[39mself\u001b[39m, \u001b[39mself\u001b[39m\u001b[39m.\u001b[39m\u001b[39m__del__\u001b[39m)\n",
      "File \u001b[0;32m~/Library/Python/3.11/lib/python/site-packages/pyChatGPT/pyChatGPT.py:184\u001b[0m, in \u001b[0;36mChatGPT.__init_browser\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    182\u001b[0m     options\u001b[39m.\u001b[39madd_argument(arg)\n\u001b[1;32m    183\u001b[0m \u001b[39mtry\u001b[39;00m:\n\u001b[0;32m--> 184\u001b[0m     \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdriver \u001b[39m=\u001b[39m uc\u001b[39m.\u001b[39;49mChrome(options\u001b[39m=\u001b[39;49moptions)\n\u001b[1;32m    185\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mTypeError\u001b[39;00m \u001b[39mas\u001b[39;00m e:\n\u001b[1;32m    186\u001b[0m     \u001b[39mif\u001b[39;00m \u001b[39mstr\u001b[39m(e) \u001b[39m==\u001b[39m \u001b[39m'\u001b[39m\u001b[39mexpected str, bytes or os.PathLike object, not NoneType\u001b[39m\u001b[39m'\u001b[39m:\n",
      "File \u001b[0;32m~/Library/Python/3.11/lib/python/site-packages/undetected_chromedriver/__init__.py:246\u001b[0m, in \u001b[0;36mChrome.__init__\u001b[0;34m(self, options, user_data_dir, driver_executable_path, browser_executable_path, port, enable_cdp_events, service_args, service_creationflags, desired_capabilities, advanced_elements, service_log_path, keep_alive, log_level, headless, version_main, patcher_force_close, suppress_welcome, use_subprocess, debug, no_sandbox, **kw)\u001b[0m\n\u001b[1;32m    240\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mdebug \u001b[39m=\u001b[39m debug\n\u001b[1;32m    241\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mpatcher \u001b[39m=\u001b[39m Patcher(\n\u001b[1;32m    242\u001b[0m     executable_path\u001b[39m=\u001b[39mdriver_executable_path,\n\u001b[1;32m    243\u001b[0m     force\u001b[39m=\u001b[39mpatcher_force_close,\n\u001b[1;32m    244\u001b[0m     version_main\u001b[39m=\u001b[39mversion_main,\n\u001b[1;32m    245\u001b[0m )\n\u001b[0;32m--> 246\u001b[0m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mpatcher\u001b[39m.\u001b[39;49mauto()\n\u001b[1;32m    247\u001b[0m \u001b[39m# self.patcher = patcher\u001b[39;00m\n\u001b[1;32m    248\u001b[0m \u001b[39mif\u001b[39;00m \u001b[39mnot\u001b[39;00m options:\n",
      "File \u001b[0;32m~/Library/Python/3.11/lib/python/site-packages/undetected_chromedriver/patcher.py:124\u001b[0m, in \u001b[0;36mPatcher.auto\u001b[0;34m(self, executable_path, force, version_main)\u001b[0m\n\u001b[1;32m    121\u001b[0m \u001b[39mexcept\u001b[39;00m \u001b[39mFileNotFoundError\u001b[39;00m:\n\u001b[1;32m    122\u001b[0m     \u001b[39mpass\u001b[39;00m\n\u001b[0;32m--> 124\u001b[0m release \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mfetch_release_number()\n\u001b[1;32m    125\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mversion_main \u001b[39m=\u001b[39m release\u001b[39m.\u001b[39mversion[\u001b[39m0\u001b[39m]\n\u001b[1;32m    126\u001b[0m \u001b[39mself\u001b[39m\u001b[39m.\u001b[39mversion_full \u001b[39m=\u001b[39m release\n",
      "File \u001b[0;32m~/Library/Python/3.11/lib/python/site-packages/undetected_chromedriver/patcher.py:145\u001b[0m, in \u001b[0;36mPatcher.fetch_release_number\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    143\u001b[0m path \u001b[39m=\u001b[39m path\u001b[39m.\u001b[39mupper()\n\u001b[1;32m    144\u001b[0m logger\u001b[39m.\u001b[39mdebug(\u001b[39m\"\u001b[39m\u001b[39mgetting release number from \u001b[39m\u001b[39m%s\u001b[39;00m\u001b[39m\"\u001b[39m \u001b[39m%\u001b[39m path)\n\u001b[0;32m--> 145\u001b[0m \u001b[39mreturn\u001b[39;00m LooseVersion(urlopen(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49murl_repo \u001b[39m+\u001b[39;49m path)\u001b[39m.\u001b[39mread()\u001b[39m.\u001b[39mdecode())\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:216\u001b[0m, in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    214\u001b[0m \u001b[39melse\u001b[39;00m:\n\u001b[1;32m    215\u001b[0m     opener \u001b[39m=\u001b[39m _opener\n\u001b[0;32m--> 216\u001b[0m \u001b[39mreturn\u001b[39;00m opener\u001b[39m.\u001b[39;49mopen(url, data, timeout)\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:519\u001b[0m, in \u001b[0;36mOpenerDirector.open\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    516\u001b[0m     req \u001b[39m=\u001b[39m meth(req)\n\u001b[1;32m    518\u001b[0m sys\u001b[39m.\u001b[39maudit(\u001b[39m'\u001b[39m\u001b[39murllib.Request\u001b[39m\u001b[39m'\u001b[39m, req\u001b[39m.\u001b[39mfull_url, req\u001b[39m.\u001b[39mdata, req\u001b[39m.\u001b[39mheaders, req\u001b[39m.\u001b[39mget_method())\n\u001b[0;32m--> 519\u001b[0m response \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_open(req, data)\n\u001b[1;32m    521\u001b[0m \u001b[39m# post-process response\u001b[39;00m\n\u001b[1;32m    522\u001b[0m meth_name \u001b[39m=\u001b[39m protocol\u001b[39m+\u001b[39m\u001b[39m\"\u001b[39m\u001b[39m_response\u001b[39m\u001b[39m\"\u001b[39m\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:536\u001b[0m, in \u001b[0;36mOpenerDirector._open\u001b[0;34m(self, req, data)\u001b[0m\n\u001b[1;32m    533\u001b[0m     \u001b[39mreturn\u001b[39;00m result\n\u001b[1;32m    535\u001b[0m protocol \u001b[39m=\u001b[39m req\u001b[39m.\u001b[39mtype\n\u001b[0;32m--> 536\u001b[0m result \u001b[39m=\u001b[39m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_call_chain(\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mhandle_open, protocol, protocol \u001b[39m+\u001b[39;49m\n\u001b[1;32m    537\u001b[0m                           \u001b[39m'\u001b[39;49m\u001b[39m_open\u001b[39;49m\u001b[39m'\u001b[39;49m, req)\n\u001b[1;32m    538\u001b[0m \u001b[39mif\u001b[39;00m result:\n\u001b[1;32m    539\u001b[0m     \u001b[39mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:496\u001b[0m, in \u001b[0;36mOpenerDirector._call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    494\u001b[0m \u001b[39mfor\u001b[39;00m handler \u001b[39min\u001b[39;00m handlers:\n\u001b[1;32m    495\u001b[0m     func \u001b[39m=\u001b[39m \u001b[39mgetattr\u001b[39m(handler, meth_name)\n\u001b[0;32m--> 496\u001b[0m     result \u001b[39m=\u001b[39m func(\u001b[39m*\u001b[39;49margs)\n\u001b[1;32m    497\u001b[0m     \u001b[39mif\u001b[39;00m result \u001b[39mis\u001b[39;00m \u001b[39mnot\u001b[39;00m \u001b[39mNone\u001b[39;00m:\n\u001b[1;32m    498\u001b[0m         \u001b[39mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:1391\u001b[0m, in \u001b[0;36mHTTPSHandler.https_open\u001b[0;34m(self, req)\u001b[0m\n\u001b[1;32m   1390\u001b[0m \u001b[39mdef\u001b[39;00m \u001b[39mhttps_open\u001b[39m(\u001b[39mself\u001b[39m, req):\n\u001b[0;32m-> 1391\u001b[0m     \u001b[39mreturn\u001b[39;00m \u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49mdo_open(http\u001b[39m.\u001b[39;49mclient\u001b[39m.\u001b[39;49mHTTPSConnection, req,\n\u001b[1;32m   1392\u001b[0m         context\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_context, check_hostname\u001b[39m=\u001b[39;49m\u001b[39mself\u001b[39;49m\u001b[39m.\u001b[39;49m_check_hostname)\n",
      "File \u001b[0;32m/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/urllib/request.py:1351\u001b[0m, in \u001b[0;36mAbstractHTTPHandler.do_open\u001b[0;34m(self, http_class, req, **http_conn_args)\u001b[0m\n\u001b[1;32m   1348\u001b[0m         h\u001b[39m.\u001b[39mrequest(req\u001b[39m.\u001b[39mget_method(), req\u001b[39m.\u001b[39mselector, req\u001b[39m.\u001b[39mdata, headers,\n\u001b[1;32m   1349\u001b[0m                   encode_chunked\u001b[39m=\u001b[39mreq\u001b[39m.\u001b[39mhas_header(\u001b[39m'\u001b[39m\u001b[39mTransfer-encoding\u001b[39m\u001b[39m'\u001b[39m))\n\u001b[1;32m   1350\u001b[0m     \u001b[39mexcept\u001b[39;00m \u001b[39mOSError\u001b[39;00m \u001b[39mas\u001b[39;00m err: \u001b[39m# timeout error\u001b[39;00m\n\u001b[0;32m-> 1351\u001b[0m         \u001b[39mraise\u001b[39;00m URLError(err)\n\u001b[1;32m   1352\u001b[0m     r \u001b[39m=\u001b[39m h\u001b[39m.\u001b[39mgetresponse()\n\u001b[1;32m   1353\u001b[0m \u001b[39mexcept\u001b[39;00m:\n",
      "\u001b[0;31mURLError\u001b[0m: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:992)>"
     ]
    }
   ],
   "source": [
    "from pyChatGPT import ChatGPT\n",
    "\n",
    "session_token = 'eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..A6rwlPIgjgI-xWGD.Bvbl71-Rkaa1-FfA9hEAV4PvoIA0P95ZaUYIzoQREaHGEXWrjP0JzTsryEH8extiZp_37AcEL2wZxmLt6-08Er7wOuddHDi6krfb1uV_ZboyM-tXEsghUkssjMehZMLeHkAZQ_tZx19IGq8U3IO1h_-nGHEfvnRs5PPrzft2dg3AmT7iTdZS36GSmS0yra1xQDGCvJwxGPiYPZEmtO5Di5Tew0nj48c4JqPqFkRISDADTJ6KL4sehnLw_2-j4eHOghnnht8sbnL8lUQ8Ytf9-hSL_GX8GwdGZEyQijfvDl4NeGlPZbmnZNryZG1_7a6h_V0jxU1Nyc1CIJ20jAn-b8joEvL7CIfXeM862IF6yC4uzXhvRNMaYYKLviVz5haB4b6t-g37ZzlMfhOR6Pev2jNeSTfuIEt4UhJoC8i5d6aoxumyd4PmsFwNrMuiCddT28cAmfuMPSW0KYE-K-Yz23k34Ua16fZuHBH9gkYfRt1zr2_zLwWygR74sOMQO2xxEEUfNcntiKVWDfaEZaS4EZJAfGEIJ-uZHI5YehabtEAgKRiVA937sYY7OZaQ12b3FntR-a_gIjQFnpX9DnnpjUlj-dLhiKK5WFSbxLgFwIoide4uu4w5LaJLPZAwGK7tLypTO7Z2TFvSyue4UF_Fn00u_nftAH4EH3TvfmvyDNZy7zkifqyupQ2FjFwjewpVTiO-l7ZLaU9F5MozpyvYwF0iH1whPrJfiyU4D0VTRIT9sNGI6hQn_C7b21F45obpZ6hdljPOUwwZR5LNdoknKtoHRL47NT-bpoAfza0ZnspuARPSrVp66c--CaJsPHGePjYSHSxECDz7Vn9ZPhJ_Nqjo_4PSWv7R0Lu_opFa9ka8kiFM4CqzgRZJjOvgzZQUNX1mw74WivCXZwAB4qyXhpywh0mIPkmxUXLOam6SopxWY9FL8673nJmIDFy4O6940auXU5mi4rLLIjqSULkLI0bOpdbZkLK5r7RBzsJXXOoYEEh0vFTYMmFjRLdVA58NKD2Pm5zOl3Lg1E45Xp7uycENFaVmO8-krLX4--CEX3I8-UjdhBvxgx2SRvdP2TRxGRC8OjVnpTSWGqnpSU-TcAjJTUBsBhQtZ4lU8zynGPzsmLPPObRAfS7WQdvrvAz0tTZSL1OJkCZnBb_NjpGzMRzwkhwJiohV76efaDSIW7KyYe_dVXL44hL783GPMGEp0L6ahMgydeWAFMLXpoaO366r56hMLyM9VlW8R3BfcmHrbnuqtIevYA5wEYtIqfLQ-gFg3ggGI8XyPg_mK4l-F0Di1OxpWcKsd3fSPspbeIEvWLFHEFwYTUWynedTwwyNY3-W15Lq8BI3uqwYI4mjKZ3nIkrS21U9AIDKVcnDVBRVj3tsFmlgNUCBAs5vykkbFSft2ksX1gyLzriHLl0aTWq2nE4OyAEhF0hUy2gZcdnctd2E02QMp4_K-KiyiOgqJHes_qKKQpenJCwKX8k_opd7zUA4BFTcca1W3R5ozLRAbLo2mQnUMuzuihU0sPZTTKnKKjTAz9roYcIvY7wpauETvGQHiypmAM_cPoxfXVMBuLZLfPq3a4t66x5ITGFcM6sjEt7Eyxo-UVzHNzxUhaAay_JccmSf0YoiAATjH4ONAGDjUleMij1nDTOS8YuZ1-x0L3C2JUDWVywCXf9Dgb2Y5_Qke_QnFl0cB-OoYNwH2vJ4mGSUHjlGZmBZDGNDGIWLXLhDiZfhNdDgeP7m7E7yQzRHVw0Hp-8R3W5Hl338ejFhjky4n0DW8SoU7aoR8GxPWBkgG7ST7KjKCHsst2iO5b6A_mkr8Mlg9GNN0x_5PaCvtcYrdPzTFPx0vRtOGjsEnlIXaOvVCEUl3dfHF7rFGV6ssYuKFJfMw94RTQhkoYNb2P7XeLJHwZK1SM8OgB3Q49r1yu0mUrS9U9stzDllLgAW6fS3xjhOMfVG4oVYmaJgz-FzOyJxSnaBigoMipLuON_nIUo7Y3Pt8nVeYunZSSYNzT0IfVbIL-HAHaMwj-kO0hmcYOmFdaRxlbiiMIqIQaCQymhXyCbNdDj5ltZBZNTSTFXrqBVt0k2ZqNSMQ_rVHV2E3p17zGZd47f6KBpkU0s9iGcp4wR1dp4bOe_Ca-zWiFt_jaQFG3YjWth55KS865mXRNcQKK4RTPEWWqRQv3yVrQUYWUojK92XOnvGpfce4wA1cPPykd8lVPNa_mSsrAUNAw-nwFjTEIdLyYKDCrGcjxUhCsnaNGi3oDbI7KrajubWmHkd0fR5E7xbylhS-QBJmGD_QmsG_gLqWttKEDWOfiCRL66ii_SYdV6LDKqSTuBYwqkJE-tDgu3qjU_RoW83Ma4Y4szNMVG-69jJDWBV1VN80f-gxWhhVW7yhvlewSVFar4yY0J4uOeNEQHQX0qV8IZMkxmMmPTTB1wwasBBq0tTxqPQ78G-VrKyhcd54E9S3REgeNl5zMPhxS1tEHMdpQQqONrSqKb4MFQGJlAYGKAZ_H4m61qcgcYwKydnPtIblLmFIwiAX_AbOJDIrpKdx_tKWyCn3Linj1kB018JgyJoYwhdYwCOurY0AaStMu6fLvjix2hlGwXFTXgzILmDCqTvLgu7nMVnxXVJnQB-TdeXYzO_QFAwRqG6JemPwiuP6r6aByv0SlboeE9aOAiRUr4DGYmIcZzRBF2xZy68UojanLKkvGKXQJyqhTlbUNx0aPHdqV8oKOKQpXxU4ob-j18Gc_xJm7WFDRZQF8_IHX82BlVyliUumPPBYOcCxSQ.2R8OfzEpLlaqUC5j5BTnSw'  # `__Secure-next-auth.session-token` cookie from https://chat.openai.com/chat\n",
    "api = ChatGPT(session_token)  # auth with session token\n",
    "api = ChatGPT(session_token, conversation_id='some-random-uuid')  # specify conversation id\n",
    "api = ChatGPT(session_token, proxy='https://proxy.example.com:8080')  # specify proxy\n",
    "api = ChatGPT(session_token, chrome_args=['--window-size=1920,768'])  # specify chrome args\n",
    "api = ChatGPT(session_token, moderation=False)  # disable moderation\n",
    "api = ChatGPT(session_token, verbose=True)  # verbose mode (print debug messages)\n",
    "\n",
    "# auth with google login\n",
    "api = ChatGPT(auth_type='google', email='example@gmail.com', password='password')\n",
    "# auth with microsoft login\n",
    "api = ChatGPT(auth_type='microsoft', email='example@gmail.com', password='password')\n",
    "# auth with openai login (captcha solving using speech-to-text engine)\n",
    "api = ChatGPT(auth_type='openai', email='example@gmail.com', password='password')\n",
    "# auth with openai login (manual captcha solving)\n",
    "api = ChatGPT(\n",
    "    auth_type='openai', captcha_solver=None,\n",
    "    email='example@gmail.com', password='password'\n",
    ")\n",
    "# auth with openai login (2captcha for captcha solving)\n",
    "api = ChatGPT(\n",
    "    auth_type='openai', captcha_solver='2captcha', solver_apikey='abc',\n",
    "    email='example@gmail.com', password='password'\n",
    ")\n",
    "# reuse cookies generated by successful login before login,\n",
    "# if `login_cookies_path` does not exist, it will process logining  with `auth_type`, and save cookies to `login_cookies_path`\n",
    "# only works when `auth_type` is `openai` or `google`\n",
    "api = ChatGPT(auth_type='openai', email='example@xxx.com', password='password',\n",
    "    login_cookies_path='your_cookies_path',\n",
    ")\n",
    "\n",
    "resp = api.send_message('Hello, world!')\n",
    "print(resp['message'])\n",
    "\n",
    "api.reset_conversation()  # reset the conversation\n",
    "api.clear_conversations()  # clear all conversations\n",
    "api.refresh_chat_page()  # refresh the chat page"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
